### PR TITLE
Add required `--mode` parameter to define sets of parameters

### LINKED ISSUE #1105: Create macros to automatically define sets of parameters based on project-specific needs
## Summary

We're going to need at least three to start with: `abcdbids` (settings used for ABCD data- perhaps specifically ABCC?), `hbcd` (settings to be used for HBCD data releases), and `linc` (settings used by the PennLINC team).

This will involve adding a required new parameter, `--mode`, that will set various parameters. Since the parameter will be required, this is a breaking change. Any additional parameters will override the ones for the "mode".

I will need to discuss the `abcdbids` and `hbcd` settings with @madisoth and @tjhendrickson.

### LINKED ISSUE #1111: Add option to disable correlation matrix creation
## Summary

Requested by Damien Fair in last Friday's meeting. The DCAN pipeline involves a secondary thresholding step before calculating correlation matrices.

### LINKED ISSUE #1129: Rename --skip-dcan-qc option
## Summary

Rename the --skip-dcan-qc option (probably to --skip-abcd-qc or --skip-abcc-qc ; confirm w/ Damien which is preferred)

### LINKED ISSUE #1130: Replace --exact-time option with --create-matrices
## Summary

- matrix output off by default; add an option --create-matrices to enable it
- if --create-matrices is used, require that user specify either to use (1) exact time or (2) all data to create the matrices

`--create-matrices {[min_time] exact_time,"all"}` (confirm if this is good w/Damien)

### LINKED ISSUE #1131: Designate certain arguments as required when a mode is used
## Summary

An options for modes to designate specific arguments as required, e.g. if using "abcd" mode require --motion-filter-type (and --band-stop-min, --band-stop-max if applicable)

## Next steps

(Need to confirm which args should be required for abcd and other modes)

### PR DESCRIPTION
**Pull request by tsalo**
Closes #1105, closes #1130, closes #1129, closes #1131, closes #1111.

## Changes proposed in this pull request
- Add required `--mode` parameter, which controls several other parameters.
    - For more detail on the actual settings for these modes, see the workflows page in the documentation (https://xcp-d--1109.org.readthedocs.build/en/1109/workflows.html#processing-modes).
- Add "y" and "n" options to flip parameters that currently rely on defaults.
- Replace `--exact-time` with `--create-matrices`.
- Add `--linc-qc` parameter.
- Replace `--skip-dcan-qc` with `--abcc-qc`.
- Rename `--combineruns` to `--combine-runs`.
- Replace `--cifti` with `--file-format`.

### CODE REVIEW

File: xcp_d/cli/parser.py [hunk 1]
```diff
@@ -845,6 +949,59 @@ def _validate_parameters(opts, build_log, parser):
...
+    if opts.exact_time == "none":
+        opts.exact_time = []
+
+    if opts.mode == "abcdbids":
+        opts.despike = False if opts.despike == "auto" else opts.despike
+        opts.cifti = True if opts.cifti == "auto" else opts.cifti
+        opts.process_surfaces = True if opts.process_surfaces == "auto" else opts.process_surfaces
+        opts.dcan_qc = True if opts.dcan_qc == "auto" else opts.dcan_qc
+        opts.exact_time = [300, 480] if opts.exact_time == "auto" else opts.exact_time
+        opts.combineruns = True if opts.combineruns == "auto" else opts.combineruns
```
Comment (tsalo) [inline xcp_d/cli/parser.py:962]: @madisoth can you weigh in on this and note any other parameters that should be set here?

File: xcp_d/cli/parser.py [hunk 2]
```diff
@@ -845,6 +949,59 @@ def _validate_parameters(opts, build_log, parser):
...
+    elif opts.mode == "hbcd":
+        opts.despike = False if opts.despike == "auto" else opts.despike
+        opts.cifti = True if opts.cifti == "auto" else opts.cifti
+        opts.process_surfaces = True if opts.process_surfaces == "auto" else opts.process_surfaces
+        opts.dcan_qc = True if opts.dcan_qc == "auto" else opts.dcan_qc
+        opts.exact_time = [300, 480] if opts.exact_time == "auto" else opts.exact_time
+        opts.combineruns = True if opts.combineruns == "auto" else opts.combineruns
```
Comment (tsalo) [inline xcp_d/cli/parser.py:969]: @tjhendrickson can you weigh in on this and note any other parameters that should be set here?

Comment (tjhendrickson) [reply xcp_d/cli/parser.py:984]: Hi @tsalo, thanks for sending along. I am not super familiar with the code base, can you break down these details into human readable form?

Comment (tsalo) [reply xcp_d/cli/parser.py:984]: Absolutely. Here's the breakdown:

1. despiking is disabled
2. CIFTI processing is enabled
3. Surfaces in fsnative space will be warped to fsLR and parcellated morphometrics (e.g., sulcal depth) will be outputtted
4. DCAN QC-related outputs (e.g., interpolated, denoised BOLD data; the BrainSprites for the executive summary; the dcan hd5 motion file) will be created
5. Correlation matrices from 5 minutes and 8 minutes will be generated
6. Results (BOLD data and parcellated time series) will be concatenated across runs

Comment (tsalo) [reply xcp_d/cli/parser.py]: Set input type to "nibabies".
Set head radius to "auto".
Set despike to True.

File: xcp_d/cli/parser.py [hunk 3]
```diff
@@ -517,32 +515,73 @@ def _bids_filter(value, parser):
...
+    g_dcan.add_argument(
+        "--linc-qc",
+        "--linc_qc",
+        action="store_true",
+        default="auto",
+        dest="linc_qc",
+        help="""\
+Run LINC QC.
+
+This will create the NiPreps-style HTML report and calculate QC metrics from the LINC pipeline.
```
Comment (tsalo) [inline xcp_d/cli/parser.py:568]:
```suggestion
This will calculate QC metrics from the LINC pipeline.
```

File: xcp_d/cli/parser.py [hunk 4]
```diff
@@ -879,6 +919,74 @@ def _validate_parameters(opts, build_log, parser):
...
+    # Check parameter value types/valid values
+    assert opts.mode in ("abcd", "hbcd", "linc"), f"Unsupported mode '{opts.mode}'."
+    assert opts.despike in (True, False, "auto")
+    assert opts.process_surfaces in (True, False, "auto")
+    assert opts.combine_runs in (True, False, "auto")
+    assert opts.file_format in ("nifti", "cifti", "auto")
+    assert opts.abcc_qc in (True, False, "auto")
+    assert opts.linc_qc in (True, False, "auto")
+
+    # Check parameters based on the mode
```
Comment (tsalo) [inline xcp_d/cli/parser.py:931]: This is the section where modes are defined.

**Comment by tsalo** [comment]
I'm going to merge this PR so that I don't have to deal with more merge conflicts, but I would still appreciate feedback on it. I will hold off on making a new release until I've gotten the 👍 from stakeholders.

**Pull request merged**